app/src/main/java/com/cris/nvh/framgiaproject/data/source/remote/TracksRemoteDataSource.java

```diff
@@ -0,0 +1,61 @@
+package com.cris.nvh.framgiaproject.data.source.remote;
+
+import android.content.Context;
+import android.os.AsyncTask;
+
+import com.cris.nvh.framgiaproject.R;
+import com.cris.nvh.framgiaproject.data.model.Genre;
+import com.cris.nvh.framgiaproject.data.model.Track;
+import com.cris.nvh.framgiaproject.data.source.TracksDataSource;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+
+/**
+ * Created by nvh
+ * Contact: toiyeuthethao1997@gmail.com
+ */
+
+public class TracksRemoteDataSource implements TracksDataSource.Remote {
+	private static TracksRemoteDataSource sRemoteTask;
+	private static Context sContext;
+
+	public TracksRemoteDataSource(Context context) {
+		sContext = context;
+	}
+
+	@Override
+	public void getListGenres(String[] urls, TracksDataSource.LoadDataCallBack<Genre> callback) {
+		new LoadTracksAsyncTask(sContext, callback).execute(urls);
+	}
+
+	@Override
+	public void getOnlineTracks(String[] urls, TracksDataSource.LoadDataCallBack<Track> callback) {
```

Comment:
Em nên load Tracks theo genre. Và có thể truyền limit và offSet vào để dùng đến đâu thì load đến đó. Em có thể update sau cũng đc.
```suggestion
	public void getOnlineTracks(String genre, int limit, int offSet, TracksDataSource.LoadDataCallBack<Track> callback) {
```